exporter/zipkin/lib/opentelemetry/exporter/zipkin/transformer.rb

```diff
@@ -0,0 +1,188 @@
+        ATTRIBUTE_NET_HOST_PORT = 'host.port'
+
+        DEFAULT_SERVICE_NAME = OpenTelemetry::SDK::Resources::Resource.default.attribute_enumerator.find { |k, _| k == SERVICE_NAME_ATTRIBUTE_KEY }&.last || 'unknown_service'
+        private_constant(:KIND_MAP, :DEFAULT_SERVICE_NAME, :SERVICE_NAME_ATTRIBUTE_KEY, :ERROR_TAG_KEY, :STATUS_CODE_NAME, :STATUS_CODE_DESCRIPTION, :STATUS_UNSET, :STATUS_ERROR, :STATUS_OK)
+
+        def to_zipkin_span(span_d, resource)
+          start_time = (span_d.start_timestamp.to_f * 1_000_000).to_i
+          duration = (span_d.end_timestamp.to_f * 1_000_000).to_i - start_time
+          tags = {}
+          service_name = DEFAULT_SERVICE_NAME
+          resource&.attribute_enumerator&.select do |key, value|
+            if key == SERVICE_NAME_ATTRIBUTE_KEY
+              service_name = value
+            else
+              tags[key] = value
+            end
+          end
+
+          add_il_tags(span_d, tags)
+          add_status_tags(span_d, tags)
+          tags = aggregate_span_tags(span_d, tags)
+
+          # TOOO: set debug flag? (is that represented in tracestate?)
+          # https://github.com/openzipkin/b3-propagation#why-is-debug-encoded-as-x-b3-flags-1
+          # https://github.com/openzipkin/zipkin-api/blob/7692ca7be4dc3be9225db550d60c4d30e6e9ec59/zipkin2-api.yaml#L475
+          # TODO: shared key mapping
+
+          zipkin_span = {
+            name: span_d.name,
+            traceId: span_d.hex_trace_id,
+            id: span_d.hex_span_id,
+            timestamp: start_time,
+            duration: duration,
+            debug: false
+          }
+
+          add_conditional_tags(zipkin_span, span_d, tags, service_name)
+          add_annotations(zipkin_span, span_d)
+
+          zipkin_span
+        end
+
+        def add_il_tags(span_data, tags)
+          tags['otel.library.name'] = span_data.instrumentation_library&.name if span_data.instrumentation_library&.name
+          tags['otel.library.version'] = span_data.instrumentation_library&.version if span_data.instrumentation_library&.version
+        end
+
+        def add_status_tags(span_data, tags)
+          if span_data.status&.code && span_data.status&.code == OpenTelemetry::Trace::Status::ERROR
+            # mark errors if we can, setting error key to description but falling back to 'true'
+            # https://github.com/open-telemetry/opentelemetry-specification/blob/84b18b23339dcc0b1a9d48f976a1afd287417602/specification/trace/sdk_exporters/zipkin.md#status
+            # https://github.com/openzipkin/zipkin-ruby/blob/7bedb4dd162c4cbeffc7b97dd06c8dbccbfbab62/lib/zipkin-tracer/trace.rb#L259
+            # https://github.com/open-telemetry/opentelemetry-collector/blob/81c7cc53b7067fbf3db4e1671b13bbe2796eb56e/translator/trace/zipkin/traces_to_zipkinv2.go#L144
+            if span_data.status&.description.nil? || span_data.status.description.empty?
+              tags[ERROR_TAG_KEY] = 'true'
```

Comment:
Spec says "use an empty string if Description has no value"
```suggestion
              tags[ERROR_TAG_KEY] = ''
```